playbooks/openstack/inventory.py

```diff
@@ -16,52 +16,72 @@
 import shade
 
 
-def base_openshift_inventory(cluster_hosts):
+def base_openshift_inventory(cluster_hosts, inventory,
+                             docker_storage_mountpoints):
     '''Set the base openshift inventory.'''
-    inventory = {}
+    nodes = set()
+    osev3 = set()
+
+    inventory['cluster_hosts'] = {'hosts': []}
+    inventory['masters'] = {'hosts': []}
+    inventory['etcd'] = {'hosts': []}
+    inventory['infra_hosts'] = {'hosts': []}
+    inventory['app'] = {'hosts': []}
+    inventory['glusterfs'] = {'hosts': []}
+    inventory['dns'] = {'hosts': []}
+    inventory['lb'] = {'hosts': []}
+    inventory['localhost'] = {'ansible_connection': 'local'}
 
-    masters = [server.name for server in cluster_hosts
-               if server.metadata['host-type'] == 'master']
+    for server in cluster_hosts:
+        if 'group' in server.metadata:
+            group = server.metadata.get('group')
+            if group not in inventory:
+                inventory[group] = {'hosts': []}
+            inventory[group]['hosts'].append(server.name)
 
-    etcd = [server.name for server in cluster_hosts
-            if server.metadata['host-type'] == 'etcd']
-    if not etcd:
-        etcd = masters
+        inventory['cluster_hosts'].append(server.name)
```

Comment:
I think this should be `inventory['cluster_hosts']['hosts'].append(server.name)`